AdyenCard/Utilities/CardType/BinInfoProvider.swift

```diff
@@ -71,13 +71,13 @@ internal final class BinInfoProvider: AnyBinInfoProvider {
         if let service = binLookupService {
             useService(service)
         } else {
-            cardPublicKeyProvider.fetch { [weak self, apiClient] result in
-                switch result {
-                case let .success(publicKey):
+            cardPublicKeyProvider.fetch { [weak self] result in
+                switch (result, self?.apiClient) {
```

Comment:
this is very cool, but a bit much i think. no way `apiClient` can be `nil`. logically we would check for `self` being `nil`. also it breaks the common pattern of switching on `Result`

Reply:
Yeah, I'd also check self nullability at the beginning